hack-sound-server

```diff
@@ -0,0 +1,99 @@
+from gi.repository import Json
+
+
+def get_datadir():
+    return os.path.join(GLib.get_user_data_dir(), "hack-sound-server")
+
+def get_metadata_path():
+    return os.path.join(get_datadir(), "metadata.json")
+
+def get_sounds_dir():
+    return os.path.join(get_datadir(), "sounds")
+
+
+class HackSoundPlayer:
+    def __init__(self, metadata, sender):
+        self.metadata = metadata
+        self.sender = sender
+        self.pipeline = self._build_pipeline()
+
+    def play(self):
+        self.pipeline.set_state(Gst.State.PLAYING)
+
+    @property
+    def sound_location(self):
+        return os.path.join(get_sounds_dir(), self.metadata["sound-file"])
+
+    def _build_pipeline(self):
+        print(self.sound_location)
+        spipeline = "filesrc location=\"%s\" ! decodebin ! autoaudiosink" % self.sound_location
+        print(spipeline)
+        return Gst.parse_launch(spipeline)
+
+
+class HackSoundServer(Gio.Application):
+
+    _DBUS_NAME = "com.endlessm.HackSoundServer"
+    _DBUS_XML = """
+    <node>
+      <interface name='com.endlessm.HackSoundServer'>
+        <method name='PlaySound'>
+          <arg type='s' name='sound_event' direction='in'/>
+          <arg type='s' name='uuid' direction='out'/>
+        </method>
+      </interface>
+    </node>
+    """
+
+    def __init__(self):
+        super().__init__(application_id=self._DBUS_NAME,
+                         flags=Gio.ApplicationFlags.IS_SERVICE)
+        self._dbus_id = None
+        with open(get_metadata_path(), "r") as metadata_file:
+            self.metadata = json.load(metadata_file)
+        self.hold()
+        self.players = {}
+
+    def do_dbus_register(self, connection, path):
+        info = Gio.DBusNodeInfo.new_for_xml(self._DBUS_XML)
+        print(connection)
+        self._dbus_id = connection.register_object(path,
+                                                   info.interfaces[0],
+                                                   self.__method_called_cb)
+        return Gio.Application.do_dbus_register(self, connection, path)
+
+    def do_dbus_unregister(self, connection, path):
+        Gio.Application.do_dbus_unregister(self, connection, path)
+        if not self._dbus_id:
+            return
+        connection.unregister_object(self._dbus_id)
+        self._dbus_id = None
+
+    def __method_called_cb(self, connection, sender, path, iface,
+                           method, params, invocation):
+        if method == "PlaySound":
+            sound_event_id = params[0]
+            uuid_ = str(uuid.uuid4())
+            self.players[uuid_] = HackSoundPlayer(self.metadata[sound_event_id],
+                                                  sender)
+            self.players[uuid_].play()
+            invocation.return_value(GLib.Variant('(s)', (uuid_, )))
+
+
+if __name__ == '__main__':
+    Gst.init(None)
+    server = HackSoundServer()
+    GLib.unix_signal_add(GLib.PRIORITY_DEFAULT, signal.SIGINT, server.quit)
```

Comment:
It would be good to add a `SIGTERM` handler that calls `server.release` as well, to quit "politely".